src/Serializer/AbstractItemNormalizer.php

```diff
@@ -420,12 +420,13 @@ protected function getAttributeValue($object, $attribute, $format = null, array
             return $this->normalizeCollectionOfRelations($propertyMetadata, $attributeValue, $className, $format, $this->createChildContext($context, $attribute));
         }
 
-        if (
-            $type &&
-            ($className = $type->getClassName()) &&
-            $this->resourceClassResolver->isResourceClass($className)
-        ) {
-            return $this->normalizeRelation($propertyMetadata, $attributeValue, $className, $format, $this->createChildContext($context, $attribute));
+        if ($type && ($className = $type->getClassName())) {
+            if ($this->resourceClassResolver->isResourceClass($className)) {
+                return $this->normalizeRelation($propertyMetadata, $attributeValue, $className, $format, $this->createChildContext($context, $attribute));
+            }
+            else {
```

Comment:
This `else` block can be removed, it is useless because of the `return` statement in the previous `if`)